### PR TITLE
linked time: moves Image thumb on range selection

### PR DESCRIPTION
**Pull request by japie1235813**
The slider thumb of image card on range selection was only moved on single selection (and there is a matched step). This pr moves the thumb on range selection based on the thumb's current position.

### CODE REVIEW

File: tensorboard/webapp/metrics/views/card_renderer/image_card_container.ts [hunk 1]
```diff
@@ -284,9 +284,30 @@ export class ImageCardContainer implements CardRenderer, OnInit, OnDestroy {
...
     ).pipe(
       map(([stepIndex, selectedTime, selectedSteps, stepValues]) => {
-        if (!selectedTime || selectedTime.endStep) return stepIndex;
+        const firstSelectedStep = selectedSteps[0];
+        const lastSelectedStep = selectedSteps[selectedSteps.length - 1];
+        if (!selectedTime || selectedTime.clipped) return stepIndex;
 
-        const nextStepIndex = stepValues.indexOf(selectedSteps[0]);
+        // Range selection
+        if (selectedTime.endStep !== null && stepIndex !== null) {
```
Comment (JamesHollyer) [inline tensorboard/webapp/metrics/views/card_renderer/image_card_container.ts:292]: What does a null stepIndex indicate? When will this happen?

File: tensorboard/webapp/metrics/views/card_renderer/image_card_container.ts [hunk 2]
```diff
@@ -284,9 +284,30 @@ export class ImageCardContainer implements CardRenderer, OnInit, OnDestroy {
       this.stepValues$
     ).pipe(
       map(([stepIndex, selectedTime, selectedSteps, stepValues]) => {
-        if (!selectedTime || selectedTime.endStep) return stepIndex;
+        const firstSelectedStep = selectedSteps[0];
```
Comment (JamesHollyer) [inline tensorboard/webapp/metrics/views/card_renderer/image_card_container.ts:287]: AFAICT this could be empty. Do we need to handle that case?

File: tensorboard/webapp/metrics/views/card_renderer/image_card_test.ts [hunk 3]
```diff
@@ -757,5 +757,83 @@ describe('image card', () => {
       slider = fixture.debugElement.query(By.css('mat-slider'));
       expect(slider.nativeElement.getAttribute('aria-valuenow')).toBe('2');
     });
+
+    describe('range selection', () => {
```
Comment (JamesHollyer) [inline tensorboard/webapp/metrics/views/card_renderer/image_card_test.ts:761]: Can you add a test for when the entire range is outside the range of available images? 
Ex:
Start step = 50
End step = 60

**Review by JamesHollyer — COMMENTED** [review]
AFAICT there is an edge case that is not handled here. If I am wrong just let me know and add the test.

File: tensorboard/webapp/metrics/views/card_renderer/image_card_test.ts (hunk 3 above)
Comment (japie1235813) [reply tensorboard/webapp/metrics/views/card_renderer/image_card_test.ts:761]: it's covered by `'dose not moves slider thumb when selected time is clipped',` I set start step 55 and end step 65

File: tensorboard/webapp/metrics/views/card_renderer/image_card_test.ts [hunk 4]
```diff
@@ -834,6 +834,16 @@ describe('image card', () => {
 
         expect(getSliderThumbPosition(fixture2)).toBe('2');
       });
+
+      it('dose not moves slider thumb when selected range had no image', () => {
```
Comment (JamesHollyer) [inline tensorboard/webapp/metrics/views/card_renderer/image_card_test.ts:838]: nit: does*

**Review by JamesHollyer — APPROVED** [review]
LGTM after nit.
